exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -93,13 +96,35 @@ public ProfileInfo(DrillConfig drillConfig, String queryId, long startTime, long
       this.time = new Date(startTime);
       this.foreman = foreman;
       this.link = generateLink(drillConfig, foreman, queryId);
-      this.query = query.substring(0,  Math.min(query.length(), 150));
+      this.query = extractQuerySnippet(query);
       this.state = state;
       this.user = user;
       this.totalCost = totalCost;
       this.queueName = queueName;
     }
 
+    private String extractQuerySnippet(String queryText) {
+      //Extract upto max char limit as snippet
+      String sizeCappedQuerySnippet = queryText.substring(0,  Math.min(queryText.length(), QUERY_SNIPPET_MAX_CHAR));
+      //Trimming down based on line-count
+      if ( QUERY_SNIPPET_MAX_LINES < sizeCappedQuerySnippet.split(System.lineSeparator()).length ) {
+        int linesConstructed = 0;
+        StringBuilder lineCappedQuerySnippet = new StringBuilder();
+        String[] queryParts = sizeCappedQuerySnippet.split(System.lineSeparator());
+        for (String qPart : queryParts) {
+          lineCappedQuerySnippet.append(qPart);
+          if ( ++linesConstructed < QUERY_SNIPPET_MAX_LINES ) {
+            lineCappedQuerySnippet.append(System.lineSeparator());
```

Comment:
Do we want to append with new line or maybe space for better readability?

Reply:
I wanted to preserve the original query format, hence, I'm applying new line instead of spaces. Besides, the snippet can also carry single-line comments, and putting them into 1 line will make it appear that anything following a `--` is all comments